### PR TITLE
TMEDIA-104 Integrated photo center search on promo blocks

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Update all promo blocks to have the new integrated photo center search feature available in PageBuilder editor

## Jira Ticket
- [TMEDIA-104](https://arcpublishing.atlassian.net/browse/TMEDIA-104)

## Acceptance Criteria

1. PageBuilder users can use the Integrated Image Search functionality when choosing images for the Image URL custom field in blocks specified above
2. The chosen image should be resized with our resizer the same way it is now (i.e. this functionality should not change) 
3. If customers have previously placed one of these blocks on a page/template and filled out an Image URL, that choice should still apply to the block (so that this update is backwards compatible) 
4. Tests and documentation are updated to cover this functionality

## Test Steps
Requires update .env settings to be added/updated in your fusion repo

```
FUSION_RELEASE=2.7.1-photo-search-beta
PB_EDITOR=rc
PB_RELEASE=3.3-photo-search-beta
```

Also requires canary version on the Fusion cli - in Fusion repo run

```npm install @arc-fusion/cli@canary```

1. Checkout this branch `git checkout TMEDIA-104-integrated-search-promo-blocks`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/extra-large-promo-block,@wpmedia/large-promo-block,@wpmedia/medium-promo-block,@wpmedia/small-promo-block,@wpmedia/extra-large-manual-promo-block,@wpmedia/large-manual-promo-block,@wpmedia/medium-manual-promo-block,@wpmedia/small-manual-promo-block`
3. From PageBuilder editor - http://localhost/pagebuilder/pages 
4. Edit Page - All Promos
5. As you hover over each promo block you will see a "Replace Image" button appear
6. Clicking on the button will open the integrated photo center search panel
7. Hovering over each image will update the editor view to show the image in the block
8. Clicking on the image in the photo center panel will set the custom field value for the image URL
9. Publishing the page with the updated image will show the new image on the published page

## Effect Of Changes

TMEDA-104-pagebuilder-editor

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps a reviewer will follow above are working.
- [X] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [X] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [X] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm run test`, made sure all tests are passing
  - [X] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [X] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx [hunk 1]
```diff
@@ -31,10 +31,11 @@ const OverlineHeader = styled.h2`
 `;
 
 const ExtraLargeManualPromoItem = ({ customFields }) => {
-  const { arcSite } = useFusionContext();
+  const { arcSite, isAdmin } = useFusionContext();
+  const { searchableField } = useEditableContent();
 
   const resizedImageOptions = useContent({
-    source: customFields.lazyLoad ? 'resize-image-api-client' : 'resize-image-api',
+    source: (customFields.lazyLoad || isAdmin) ? 'resize-image-api-client' : 'resize-image-api',
```
Comment (brentmiller1973) [inline blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:38]: I know this line hasn't changed much with this PR, but could you add a line or two of comments what this is doing?  ust want to have some context here for new dev and if we come back to this after several months and forget whats going on :-)

**Review by brentmiller1973 — CHANGES_REQUESTED** [review]
@matthewroach Code looks solid.  Just asking for some code comments to make clear how this is coming together.  

Now during testing....Not sure if this is on my end or not, but while I can get the side bar to show up....it never loads images:  

image

I followed your setup instructions.  I had an existing page so I added an extra large promo to it.  The `Replace Image` button shows...when I click that the panel comes up fine but never loads images--just keeps spinning (do I need to be on a VPN for this?).  If it's not a VPN thing, might want to get another dev to test this out locally to make sure.  I do see the following error in the console:  `Error: Could not find source: resize-image-api-client`. Also the image url that I initially used was `https://cloudfront-us-east-1.images.arcpublishing.com/corecomponents/WAHK3LTYLJGPJBOGU4UHD7IXZQ.jpg` which should be showing up.

**Comment by matthewroach** [comment]
@visionbegin - I double checked my notes, and looks like a missed a step - I updated the original comment, but you need to use a canary version of the cli - in the Fusion repo run `npm install @arc-fusion/cli@canary`

**Review by brentmiller1973 — APPROVED** [review]
Works now with the canary CLI....cool stuff!

**Pull request merged**